MoreLinq/RandomSubset.cs

```diff
@@ -61,30 +61,29 @@ public static IEnumerable<T> RandomSubset<T>(this IEnumerable<T> source, int sub
             if (source == null) throw new ArgumentNullException(nameof(source));
             if (subsetSize < 0) throw new ArgumentOutOfRangeException(nameof(subsetSize));
 
-            return RandomSubsetImpl(source, rand, seq => (seq.ToArray(), subsetSize));
+            return RandomSubsetImpl(source, rand, subsetSize);
         }
 
-        static IEnumerable<T> RandomSubsetImpl<T>(IEnumerable<T> source, Random rand, Func<IEnumerable<T>, (T[], int)> seeder)
+        static IEnumerable<T> RandomSubsetImpl<T>(IEnumerable<T> source, Random rand, int? subsetSize)
         {
             // The simplest and most efficient way to return a random subset is to perform
             // an in-place, partial Fisher-Yates shuffle of the sequence. While we could do
             // a full shuffle, it would be wasteful in the cases where subsetSize is shorter
             // than the length of the sequence.
             // See: http://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle
 
-            var (array, subsetSize) = seeder(source);
+            var array = source.ToList();
```

Comment:
`ToList()` requires one less allocation than `ToArray()`

Reply:
This change is unrelated to the subject of this PR, so please submit it in a separate one.

Reply:
Reverted with 93ca56b33dcb45736e91074c2403e62d0362fa51.